src/PhpErrorController.php

```
@@ -88,19 +89,53 @@ public static function factoryThrowableLog(\Throwable $throwable, ?string $messa
      * @param LogActionUpdater $updater
      * @return PhpErrorController
      */
-    public static function new(bool $logSilencedErrors, LogActionUpdater $updater): PhpErrorController
-    {
-        return new self($logSilencedErrors, $updater);
+    public static function new(
+        int $errorTypes,
+        bool $logExceptions,
+        bool $logSilencedErrors,
+        LogActionUpdater $updater
+    ): PhpErrorController {
+
+        return new self($errorTypes, $logExceptions, $logSilencedErrors, $updater);
     }
 
     /**
+     * @param int $errorTypes
+     * @param bool $logExceptions
      * @param bool $logSilencedErrors
      * @param LogActionUpdater $updater
      */
-    private function __construct(bool $logSilencedErrors, LogActionUpdater $updater)
+    private function __construct(
+        private readonly int $errorTypes,
+        private readonly bool $logExceptions,
+        private readonly bool $logSilencedErrors,
+        private readonly LogActionUpdater $updater
+    ) {
+    }
+
+    /**
+     * @return void
+     */
+    public function setup(): void
     {
-        $this->logSilencedErrors = $logSilencedErrors;
-        $this->updater = $updater;
+        if ($this->alreadySetup) {
+            throw new \Exception(__METHOD__ . ' can only be executed once.');
+        }
```

Should we really throw an exception here, or just return?

---

The idea is that consumers should never call this method. It is Wonolog itself that calls it. So if someone calls it, I wanted to make they know very soon that they are doing something wrong. Reurning is also fine if you prefer that.

---

Ok, I marked the method as `@private` and then return not throw.